journal/src/main/java/io/zeebe/journal/file/ChecksumGenerator.java

```diff
@@ -23,9 +23,10 @@
   private final CRC32 crc32 = new CRC32();
 
   /** Compute checksum of given ByteBuffer */
-  public long compute(final ByteBuffer buffer) {
+  public long compute(final ByteBuffer buffer, final int offset, final int length) {
+    buffer.position(offset);
     crc32.reset();
-    crc32.update(buffer);
+    crc32.update(buffer.slice().limit(length));
```

Comment:
As this is shared by both reader/writer, and we don't want to mutate, I would propose using `asReadOnlyBuffer` instead of slice - it means you need to set the position after, e.g. `buffer.asReadOnlyBuffer().position(offset).limit(offset + limit)`

IIRC limit is the max offset, not the length :sweat_smile: